src/lua/httpc.lua

```diff
@@ -459,13 +584,22 @@ curl_mt = {
                 resp.headers = process_headers(resp.headers)
             end
 
-            resp.url = url_with_params
-
-            return setmetatable(resp, {
-                __index = {
-                    decode = decode_body,
-                },
-            })
+            if not opts.chunked then
+                resp.url = url_with_params
+                return setmetatable(resp, {
+                    __index = {
+                        decode = decode_body,
+                    },
+                })
+            else
+                return setmetatable(resp, {
+                    __index = {
```

Comment:
It might be nice to add some flag like `is_stream` to distinguish a different type of  `resp` in client code.